cmd/util/app.go

```diff
@@ -623,36 +623,62 @@ func constructAppsBaseOnName(appName string, labels, annotations, args []string,
 	}, nil
 }
 
-func constructAppsFromFileUrl(fileURL, appName string, labels, annotations, args []string, appOpts AppOptions, flags *pflag.FlagSet) ([]*argoappv1.Application, error) {
+func constructAppsFromFileUrl(fileURL, appName string, labels, annotations, args []string, appOpts AppOptions, flags *pflag.FlagSet, filterByAppName bool) ([]*argoappv1.Application, error) {
 	apps := make([]*argoappv1.Application, 0)
 	// read uri
 	err := readAppsFromURI(fileURL, &apps)
 	if err != nil {
 		return nil, err
 	}
+	filteredApps := make([]*argoappv1.Application, 0)
 	for _, app := range apps {
-		if len(args) == 1 && args[0] != app.Name {
-			return nil, fmt.Errorf("app name '%s' does not match app spec metadata.name '%s'", args[0], app.Name)
-		}
-		if appName != "" && appName != app.Name {
-			app.Name = appName
+		if filterByAppName {
+			if len(args) == 1 {
+				if appName != "" && appName != args[0] {
+					return nil, fmt.Errorf("--name argument '%s' does not match app name %s", appName, args[0])
+				}
+				appName = args[0]
+			}
+
+			if appName == app.Name {
```

Comment:
Why do we need a separate `filterByAppName`?

Reply:
I think in case the file given as `--local-app-manifest` contains multiple YAML documents.